src/dropdown/dropdown.js

```diff
@@ -170,9 +162,50 @@ angular.module('ui.bootstrap.dropdown', ['ui.bootstrap.position'])
         return;
       }
       dropdownCtrl.dropdownMenu = element;
+
+      var dropdownMenu = element,
+          dropdown = element.parent(),
+          appendToBody = dropdown.attr('dropdown-append-to-body');
+
+      scope.$watch(dropdownCtrl.isOpen, function( isOpen ) {
+        if ( isOpen ) {
+
+          // Add invisible dropdownMenu to DOM to calculate its height
+          dropdownMenu.css({
+            visibility: 'hidden',
+            display:    'block'
+          });
+
+          var windowBottom = $window.innerHeight + $window.pageYOffset,
+              dropdownOffset = $position.offset(dropdown).top,
+              dropdownHeight = $position.position(dropdown).height,
+              dropdownMenuHeight = $position.position(dropdownMenu).height,
+              dropdownMenuBottom = dropdownOffset + dropdownHeight + dropdownMenuHeight;
+
+          if ( appendToBody !== undefined ) {
```

Comment:
Here, if you check for falseness `dropdown.attr('dropdown-append-to-body')` evaluates to `false` even if `dropdown-append-to-body` attribute is applied to dropdown element. 
So we need to specifically target its `undefined` state.